scripts/ci/check_license.py

```diff
@@ -0,0 +1,304 @@
+            self.write_junit()
+            sys.exit(2)
+        stdout, stderr = process.communicate()
+        stdout = stdout.decode('utf-8')
+        stderr = stderr.decode('utf-8')
+        if process.returncode or stderr:
+            self.report('error', f'Command "{run_str}" exited with {process.returncode}\n'
+                                 f'==stdout==\n{stdout}\n==stderr==\n{stderr}')
+            self.write_junit()
+            sys.exit(2)
+        return stdout.rstrip()
+
+    def report(self, label: str, message: str, file_name: 'str|Path' = '<none>', license: str = ''):
+        '''Report results to the user.'''
+        file_name = str(file_name)
+        # Print to stdout/stderr with optional GitHub Actions Workflow commands.
+        if not self.args.github:
+            if label in ('error', 'warning'):
+                print(f'{label.upper()}: {file_name}: {message}', file=sys.stderr)
+            else:
+                print(f'{label.upper()}: {file_name}: {message}')
+        else:
+            print(f'{file_name}: ')
+            if label in ('error', 'warning'):
+                if file_name != '<none>':
+                    file_path = (self.west_workspace / file_name).relative_to(self.git_top)
+                else:
+                    file_path = file_name
+                print(f'::{label} file={file_path},title=License Problem::' +
+                      message.replace('%', '%25').replace('\r', '%0D').replace('\n', '%0A'))
+            else:
+                print(f'{label.upper()}: {message}')
+        # Put result in JUnit file.
+        test_case = junit_xml.TestCase(file_name + (f':{license}' if license else ''),
+                                       'LicenseCheck')
+        if label == 'error':
+            test_case.add_failure_info(message)
+        elif label == 'warning':
+            test_case.add_skipped_info('WARNING: ' + message)
+        elif label == 'skip':
+            test_case.add_skipped_info(message)
+        self.junit_test_cases.append(test_case)
+        # Increment counters.
+        self.total_tests += 1
+        if label == 'error':
+            self.total_errors += 1
+        elif label == 'warning':
+            self.total_warnings += 1
+        elif label == 'skip':
+            self.total_skipped += 1
+
+    def generate_list_of_files(self) -> 'list[Path]':
+        '''
+        Generate a list of files changed in the git patch. The returned path is relative to the
+        west workspace.
+        '''
+        files = self.run('git', 'diff', '--name-only', '--diff-filter=d', self.args.commits)
+        files = files.splitlines()
+        files = [(self.git_top / f.strip()).relative_to(self.west_workspace)
+                 for f in files if f.strip()]
+        return files
+
+    def skip_files(self, files: 'list[Path]') -> 'list[Path]':
+        '''
+        Skip files that will not be checked because they are not exist or any license
+        is allowed in them.
```

Comment:
This is grammatically incorrect and weird otherwise as well. If this is visible somewhere, fix this, for example something like this:

..., because they do not exist, or allow a a license.